### ISSUE TITLE
FCM - didReceiveRegistrationToken broken

### ISSUE THREAD

**Issue by iwwwh** [firebase/firebase-ios-sdk#13596]
### Description

Hi,

FCM delegate method didReceiveRegistrationToken is not being called on a first app launch until user gives APNS permission, if permission is not granted it won’t ever be called until the next app launch.

Steps to reproduce:

1. Prepare fresh iOS simulator (Erase All Content and Settings). In our tests it was iPhone 15 Pro iOS 17.5.
2. Launch an app there with FCM installed and configured with Swizzling enabled. Before that add some logging to didReceiveRegistrationToken.
3. See that didReceiveRegistrationToken never triggered.

Note: If we delete the app on the same device, without erasing the device, and then install the app again - everything works normally. So the issue can be reproduced only on a fresh devices.

The same we observe on production: all of our new users are without FCM tokens until they restart the app.

### Firebase SDK Version

10.29.0

### Xcode Version

15.4

### Installation Method

Swift Package Manager

### Firebase Product(s)

Messaging

### Targeted Platforms

iOS

**Comment by iwwwh** [comment]
After more testing with and without Swizzling we see that FCM’s didReceiveRegistrationToken being called only after user gives APNS permission and APNS token is set (Messaging.messaging().apnsToken = …)

I’m not sure that this is expected behaviour, because the FCM docs says following:

_By default, the FCM SDK generates a registration token for the client app instance on **app launch**. Similar to the APNs device token, this token allows you to send targeted notifications to any particular instance of your app._
(https://firebase.google.com/docs/cloud-messaging/ios/client#access_the_registration_token)

Considering this we are expecting FCM token on app launch, regardless of APNS permission.

**Comment by rizafran** [comment]
Hi @iwwwh, it seems like this behavior has been changed since [v10.6](https://firebase.google.com/support/release-notes/ios#fcm_7) due to the error in fetching the APNS token. You may check the discussion [here](https://github.com/firebase/firebase-ios-sdk/issues/10679#issuecomment-1402776795) for more info.

**Comment by iwwwh** [comment]
Hi @rizafran , thanks for the link provided, that’s exactly what we are observing and what we needed to understand. I guess a documentation may need some correction to point out that token won’t be generated on the app launch, but only after APNS token provided.

One more question if you don’t mind: we’ve used FCM for many years with standard setup with Swizzling enabled; today, just for testing purposes, we disabled Swizzling and made necessary changes for Firebase products to work without it. Should we leave Swizzling disabled? I guess it makes more sense because now everything is clear and visible, no hidden magic code. What’s a recommended approach?
